src/qtAliceVisionImageIO/QtAliceVisionImageIOHandler.cpp

```diff
@@ -0,0 +1,261 @@
+                    if(isDepthMap)
+                        color = getColor32fFromJetColorMap(normalizedDepthValue);
+                    else if(isNmodMap)
+                        color = getColor32fFromJetColorMapClamp(normalizedDepthValue);
+                    tmpBuf.setpixel(x, y, color.m, 3); // set only 3 channels (RGB)
+                }
+            }
+        }
+        else
+        {
+#pragma omp parallel for
+            for(int y = 0; y < inSpec.height; ++y)
+            {
+                for(int x = 0; x < inSpec.width; ++x)
+                {
+                    float depthValue = 0.0f;
+                    inBuf.getpixel(x, y, &depthValue, 1);
+                    Color32f color = getColor32fFromJetColorMap(depthValue);
+                    tmpBuf.setpixel(x, y, color.m, 3); // set only 3 channels (RGB)
+                }
+            }
+        }
+        qDebug() << "[QtAliceVisionImageIO] compute colormap done";
+        inBuf.swap(tmpBuf);
+    }
+    // Shuffle channels to convert from OIIO to Qt
+    else
+    {
+        qDebug() << "[QtAliceVisionImageIO] shuffle channels";
+
+        int channelOrder[] = {2, 1, 0, -1};
+        float channelValues[] = {1.f, 1.f, 1.f, 1.f};
+        oiio::ImageBufAlgo::channels(tmpBuf, inBuf, 4, channelOrder, channelValues, {}, false);
+        inBuf.swap(tmpBuf);
+    }
+    
+    qDebug() << "[QtAliceVisionImageIO] fill output QImage";
+    inBuf.get_pixels(exportROI, typeDesc, result.bits());
+
+    if (pixelAspectRatio != 1.0f)
+    {
+        QSize newSize(inSpec.width * pixelAspectRatio, inSpec.height);
+        result = result.scaled(newSize, Qt::IgnoreAspectRatio, Qt::SmoothTransformation);
+    }
+
+    if (_scaledSize.isValid())
+    {
+        qDebug() << "[QtAliceVisionImageIO] _scaledSize: " << _scaledSize.width() << "x" << _scaledSize.height();
+        *image = result.scaled(_scaledSize, Qt::KeepAspectRatio, Qt::SmoothTransformation);
+    }
+    else
+    {
+        *image = result;
+    }
+    return true;
+}
+
+bool QtAliceVisionImageIOHandler::write(const QImage &image)
+{
+    // TODO
+    return false;
+}
+
+bool QtAliceVisionImageIOHandler::supportsOption(ImageOption option) const
+{
+    if(option == Size)
+        return true;
+    if(option == ImageTransformation)
+        return true;
+    if(option == ScaledSize)
+        return true;
+
+    return false;
+}
+
+QVariant QtAliceVisionImageIOHandler::option(ImageOption option) const
+{
+    QFileDevice* d = dynamic_cast<QFileDevice*>(device());
+    if(!d)
+    {
+        qDebug("[QtAliceVisionImageIO] Read image failed (not a FileDevice).");
+        return QImageIOHandler::option(option);
+    }
+    std::string path = d->fileName().toStdString();
+    oiio::ImageSpec spec = aliceVision::image::readImageSpec(path);
+
+    if (option == Size)
+    {
+        return QSize(spec.width, spec.height);
+    }
+    else if(option == ImageTransformation)
+    {
+        int orientation = 0;
```

Comment:
coordinate with Float viewer ?
No exif in both or Exif in both ?

Reply:
@fabiencastan shouldn't this be directly handled at the `readImage` level (for example with a dedicated read option) ?

Reply:
Since this image IO plugin simply returns a QImage, we can't pass the orientation info, so we have to apply the transformation here.
However for the FloatImageViewer the orientation is passed so it should be applied at display time (i.e in Meshroom's QML code).

Reply:
Could be managed on qml side with autoTransform:
https://doc.qt.io/qt-6/qml-qtquick-image.html#autoTransform-prop